### ISSUE TITLE
Failed to legalize operation 'tosa.rescale'

### ISSUE THREAD

**Issue by mariecwhite** [iree-org/iree#10291]
### What happened?

When compiling SSD-MobileNetV2 with dynamic shapes, geting error:

```
ssd_mobilenet_v2_dynamic_1.0_int8.mlir:185:12: error: failed to legalize operation 'tosa.rescale'
    %182 = "tosa.rescale"(%181) {double_round = true, input_zp = 0 : i32, multiplier = [1376784203 : i32], output_zp = 0 : i32, per_channel = false, scale32 = true, shift = [38 : i32]} : (tensor<1x?x?x32xi32>) -> tensor<1x?x?x32xi8>
           ^
ssd_mobilenet_v2_dynamic_1.0_int8.mlir:185:12: note: see current operation: %224 = "tosa.rescale"(%223) {double_round = true, input_zp = 0 : i32, multiplier = [1376784203 : i32], output_zp = 0 : i32, per_channel = false, scale32 = true, shift = [38 : i32]} : (tensor<1x?x?x32xi32>) -> tensor<1x?x?x32xi8>
compilation failed
```

TFLite and MLIR file: https://storage.googleapis.com/iree-model-artifacts/ssd_mobilenet_v2_dynamic_1.0_int8.zip

**Comment by NatashaKnk** [comment]
Should be fixed with [D133383](https://reviews.llvm.org/D133383), can you verify?

**Comment by rsuderman** [comment]
Verified it compiled safely.
